components/kyma-environment-broker/internal/storage/encrypt.go

```diff
@@ -113,32 +89,17 @@ func (e *Encrypter) EncryptSMCreds(pp *internal.ProvisioningParameters) error {
 			XSAppName:         creds.XSAppName,
 		}
 	}
-	pp.ErsContext.ServiceManager = encrypted.ServiceManager
 	pp.ErsContext.SMOperatorCredentials = encrypted.SMOperatorCredentials
 	return nil
 }
 
 func (e *Encrypter) DecryptSMCreds(pp *internal.ProvisioningParameters) error {
-	if pp.ErsContext.ServiceManager == nil && pp.ErsContext.SMOperatorCredentials == nil {
+	if pp.ErsContext.SMOperatorCredentials == nil {
 		return nil
 	}
 	var err error
-	var username, password, clientID, clientSecret []byte
-	if pp.ErsContext.ServiceManager != nil {
-		creds := pp.ErsContext.ServiceManager.Credentials.BasicAuth
-		if creds.Username != "" {
-			username, err = e.Decrypt([]byte(creds.Username))
-			if err != nil {
-				return errors.Wrap(err, "while decrypting username")
-			}
-		}
-		if creds.Password != "" {
-			password, err = e.Decrypt([]byte(creds.Password))
-			if err != nil {
-				return errors.Wrap(err, "while decrypting password")
-			}
-		}
-	}
+	var clientID, clientSecret []byte
+
 	if pp.ErsContext.SMOperatorCredentials != nil {
```

Comment:
nit: I think this check is no longer necessary because on line 97 KEB will short-circuit
https://github.com/kyma-project/control-plane/blob/b5fd6f562b159bce522b9f83a3d7e9216ca0e493/components/kyma-environment-broker/internal/storage/encrypt.go#L97-L99